// @thirdweb-dev/react/solana/dist/thirdweb-dev-react-solana.cjs.dev.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var solana = require('@thirdweb-dev/solana');
var React = require('react');
var jsxRuntime = require('react/jsx-runtime');
var walletAdapterReact = require('@solana/wallet-adapter-react');
var walletAdapterWallets = require('@solana/wallet-adapter-wallets');

/**
 * Gives access to the ThirdwebSDK instance and other useful hooks to the rest of the app.
 * Requires to be wrapped with a ConnectionProvider and a WalletProvider from @solana/wallet-adapter-react.
 * @example
 * ```tsx
 * import {
 *   useConnection,
 *   useWallet,
 * } from "@solana/wallet-adapter-react";
 * import { ThirdwebProvider } from "@thirdweb-dev/react/solana";
 *
 * const ThirdwebApp = () => {
 *  const { connection } = useConnection();
 *  const wallet = useWallet();
 *  return (
 *    <ThirdwebSDKProvider connection={connection} wallet={wallet}>
 *      <YourApp />
 *    </ThirdwebSDKProvider>
 * )};
 * ```
 */
const ThirdwebSDKProvider = _ref => {
  let {
    children,
    connection,
    wallet
  } = _ref;
  const [sdk, setSDK] = React.useState(null);
  React.useEffect(() => {
    if (connection) {
      setSDK(new solana.ThirdwebSDK(connection));
    } else {
      setSDK(null);
    }
  }, [connection]);
  React.useEffect(() => {
    if (sdk) {
      if (wallet !== null && wallet !== void 0 && wallet.publicKey) {
        sdk.wallet.connect(wallet);
      } else {
        sdk.wallet.disconnect();
      }
    }
  }, [sdk, wallet]);
  return /*#__PURE__*/jsxRuntime.jsx(ThirdwebSDKContext.Provider, {
    value: sdk,
    children: children
  });
};
const ThirdwebSDKContext = /*#__PURE__*/React.createContext(null);

const DEFAULT_WALLETS = [new walletAdapterWallets.PhantomWalletAdapter()];
/**
 * Gives access to the ThirdwebSDK instance and other useful hooks to the rest of the app.
 * Requires to be wrapped with a ConnectionProvider and a WalletProvider from @solana/wallet-adapter-react.
 * @example
 * ```tsx
 * import { ThirdwebProvider } from "@thirdweb-dev/react/solana";
 *
 * const App = () => {
 *  return (
 *     <ThirdwebProvider endpoint="devnet">
 *       <YourApp />
 *     </ThirdwebProvider>
 * )};
 * ```
 * @beta
 */

const ThirdwebProvider = _ref => {
  let {
    endpoint,
    wallets,
    autoConnect,
    children
  } = _ref;
  const clusterUrl = solana.getUrlForNetwork(endpoint);
  return /*#__PURE__*/jsxRuntime.jsx(walletAdapterReact.ConnectionProvider, {
    endpoint: clusterUrl,
    children: /*#__PURE__*/jsxRuntime.jsx(walletAdapterReact.WalletProvider, {
      wallets: wallets || DEFAULT_WALLETS,
      autoConnect: autoConnect || true,
      children: /*#__PURE__*/jsxRuntime.jsx(ThirdwebWrapperProvider, {
        children: children
      })
    })
  });
};
/**
 * @internal
 */

const ThirdwebWrapperProvider = _ref2 => {
  let {
    children
  } = _ref2;
  const {
    connection
  } = walletAdapterReact.useConnection();
  const wallet = walletAdapterReact.useWallet();
  return /*#__PURE__*/jsxRuntime.jsx(ThirdwebSDKProvider, {
    connection: connection,
    wallet: wallet,
    children: children
  });
};

function useSDK() {
  return React.useContext(ThirdwebSDKContext);
}

exports.ThirdwebProvider = ThirdwebProvider;
exports.ThirdwebSDKContext = ThirdwebSDKContext;
exports.ThirdwebSDKProvider = ThirdwebSDKProvider;
exports.ThirdwebWrapperProvider = ThirdwebWrapperProvider;
exports.useSDK = useSDK;
